tools/ncbi_blast_plus/ncbi_macros.xml

```diff
@@ -108,18 +111,169 @@
+                    <option value="-gapopen 19 -gapextend 1">Existence: 19  Extension: 1</option>
+                    <option value="-gapopen 18 -gapextend 1">Existence: 18  Extension: 1</option>
+                    <option value="-gapopen 17 -gapextend 1">Existence: 17  Extension: 1</option>
+                </param>
+
+            </when>
+            <when value="PAM70">
+                <param name="gap_costs" type="select" label="Gap Costs">
+                    <option value="-gapopen 8 -gapextend 2">Existence: 8  Extension: 2</option>
+                    <option value="-gapopen 7 -gapextend 2">Existence: 7  Extension: 2</option>
+                    <option value="-gapopen 6 -gapextend 2">Existence: 6  Extension: 2</option>
+                    <option value="-gapopen 11 -gapextend 1">Existence: 11  Extension: 1</option>
+                    <option value="-gapopen 10 -gapextend 1" selected="true">Existence: 10  Extension: 1 (default)</option>
+                    <option value="-gapopen 9 -gapextend 1">Existence: 9  Extension: 1</option>
+                </param>
+
+            </when>
+            <when value="PAM30">
+                <param name="gap_costs" type="select" label="Gap Costs">
+                    <option value="-gapopen 7 -gapextend 2">Existence: 7  Extension: 2</option>
+                    <option value="-gapopen 6 -gapextend 2">Existence: 6  Extension: 2</option>
+                    <option value="-gapopen 5 -gapextend 2">Existence: 5  Extension: 2</option>
+                    <option value="-gapopen 10 -gapextend 1">Existence: 10  Extension: 1</option>
+                    <option value="-gapopen 9 -gapextend 1" selected="true">Existence: 9  Extension: 1 (default)</option>
+                    <option value="-gapopen 8 -gapextend 1">Existence: 8  Extension: 1</option>
+                </param>
+            </when>
+        </conditional>
+    </xml>
+
+    <!-- these are still used and needed for blastn, tblastx -->
+    <!-- otherwise, the combined matrix/gapcosts macro is used -->
+    <xml name="input_gapopen">
+        <param argument="-gapopen" type="integer" optional="true" min="1"
+            label="Cost to open a gap"
+            help="Leave blank for default. See tool help for defaults."/>
+    </xml>
+
+    <xml name="input_gapextend">
+        <param argument="-gapextend" type="integer" optional="true" min="1"
+            label="Cost to extend a gap"
+            help="Leave blank for default. See tool help for defaults."/>
+    </xml>
+
     <xml name="input_scoring_matrix">
         <param argument="-matrix" type="select" label="Scoring matrix">
+            <option value="" selected="true">Use default</option>
             <option value="BLOSUM90">BLOSUM90</option>
             <option value="BLOSUM80">BLOSUM80</option>
-            <option value="BLOSUM62" selected="true">BLOSUM62 (default)</option>
+            <option value="BLOSUM62">BLOSUM62 (default)</option>
```

Comment:
I think it would be better to revert the changes to this macro, since basically the default here is always BLOSUM62 and the empty value breaks the command of `ncbi_makeprofiledb.xml`

Reply:
Checking ``makeprofiledb``, it too has ``-matrix`` with ``BLOSUM62`` as the default:

```
 -matrix <String, `BLOSUM45', `BLOSUM50', `BLOSUM62', `BLOSUM80', `BLOSUM90',
                  `PAM250', `PAM30', `PAM70'>
   Scoring matrix name
   Default = `BLOSUM62'
```

So alternatively we'd have to update the ``<command>`` to handling this with an in statement, currently it uses:

```
#if str($contain_pssm_scores.contain_pssm_scores_type) == 'no':
    -gapopen $contain_pssm_scores.gapopen
    -gapextend $contain_pssm_scores.gapextend
    -scale $contain_pssm_scores.scale
    -matrix $contain_pssm_scores.matrix
#end if
```

Reply:
The matrix default is almost always BLOSUM62 - the one exception is important in the `blastp` wrapper, because if someone chooses `blastp-short` and then opens advanced options, it looks like the default is BLOSUM62 when it should be PAM30. A small thing, but it comes up fairly often.

Reply:
Ah, I realized that the `input_scoring_matrix` macro is only used in situations where BLOSUM62 is indeed the default, so I reverted it.

Reply:
If necessary, maybe add a comment to this macro warning about those cases where BLOSUM62 is not the default?